### PR TITLE
🏃 Migrate CI to Wallaby

### PR DESCRIPTION
**Pull request by stephenfin**
**What this PR does / why we need it**:

Migrate the CI to use the Wallaby release. This is an incremental change to get
us up to the latest stable release, Xena.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Fixes: #1092

**Special notes for your reviewer**:

No image version changes.

**TODOs**:

- [ ] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

**Comment by tobiasgiese** [comment]
Please create an issue to track this change, especially regarding the next step to Xena

And thanks for contributing 🙂

**Comment by stephenfin** [comment]
> Please create an issue to track this change, especially regarding the next step to Xena
> 
> And thanks for contributing slightly_smiling_face

Done. Note that I intentionally haven't squashed commits here since there are a couple of related but separate things going on. Let me know if these need to be squashed regardless or if they must be separate PRs.

**Comment by tobiasgiese** [comment]
> Done. Note that I intentionally haven't squashed commits here since there are a couple of related but separate things going on. Let me know if these need to be squashed regardless or if they must be separate PRs.

I'm totally fine with it 👍🏻 nice work
Do you mind updating the [README.md](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/30199f697f58e034a21e1dd58016521335ddbb2e/README.md?plain=1#L59-L64) regarding our new supported OpenStack version?

**Comment by stephenfin** [comment]
> > Done. Note that I intentionally haven't squashed commits here since there are a couple of related but separate things going on. Let me know if these need to be squashed regardless or if they must be separate PRs.
> 
> I'm totally fine with it 👍🏻 nice work Do you mind updating the [README.md](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/30199f697f58e034a21e1dd58016521335ddbb2e/README.md?plain=1#L59-L64) regarding ~our~ this new supported/tested OpenStack version (v1beta1 is the recent `main` branch)?

Sure, done [*]

[*] I think. I assume we can't state support for Wallaby with older releases since we don't test those on Wallaby yet (if we ever will)

**Comment by jichenjc** [comment]
overall looks good , thanks for the update 

/lgtm

**Comment by tobiasgiese** [comment]
> I think. I assume we can't state support for Wallaby with older releases since we don't test those on Wallaby yet (if we ever will)

Maybe it's okay to create cherry-pick PRs for our release branches and test the OpenStack versions with our e2e tests.
WDYT @jichenjc?

**Comment by jichenjc** [comment]
>>>Maybe it's okay to create cherry-pick PRs for our release branches and test the OpenStack versions with our e2e tests.

agree :) I think it's reasonable ,technically update to W seems should be not be impacted for release branches.

**Comment by stephenfin** [comment]
Sounds good. I'll attempt the backport once this merges and update the `README` as the backports are merged. Anything else I need to do in the interim to get this landed?

As an aside, tested Xena today and it works as expected. [I needed to backport a patch for DevStack](https://review.opendev.org/c/openstack/devstack/+/821908) but once that lands I should be able to update this to Xena proper.

File: hack/ci/cloud-init/controller.yaml.tpl
Comment (macaptain) [inline hack/ci/cloud-init/controller.yaml.tpl:37]: Thanks for fixing this comment.

File: hack/ci/cloud-init/controller.yaml.tpl
Comment (macaptain) [inline hack/ci/cloud-init/controller.yaml.tpl:35]: I guess the Nova object store and Glance registry which were removed are gone in Wallaby? We weren't using those services anyway, so good to have them removed.

**Review by macaptain — APPROVED** [review]
Thanks for the contribution!

/lgtm

File: hack/ci/cloud-init/controller.yaml.tpl
Comment (stephenfin) [reply hack/ci/cloud-init/controller.yaml.tpl:35]: The glance registry was deprecated in Train and removed in Wallaby, I think. Nova object store has been removed since forever

**Comment by stephenfin** [comment]
Anything else you need from me to get this merged? I have the Xena change lined up waiting on this to merge first.

**Pull request merged**
